gonotego/settings/wifi.py

```diff
@@ -21,79 +21,176 @@ def save_networks(networks):
-    if network.get('psk'):
-      network_config = f"""network={{
-        ssid="{network['ssid']}"
-        psk="{network['psk']}"
-        key_mgmt=WPA-PSK
-}}"""
-    else:
-      network_config = f"""network={{
-        ssid="{network['ssid']}"
-        key_mgmt=NONE
-}}"""
-    network_configs.append(network_config)
-  
-  managed_config = "# BEGIN Go Note Go managed WiFi networks\n"
-  if networks:
-    managed_config += "\n" + "\n".join(network_configs) + "\n"
-  managed_config += "# END Go Note Go managed WiFi networks"
-  
-  # Read the existing config
+
+
+def get_gonote_managed_connections():
+  """Get list of WiFi connections currently managed by Go Note Go."""
   try:
-    with open(filepath, 'r') as f:
-      config = f.read()
+    result = subprocess.run(
+      ["nmcli", "-t", "-f", "NAME,TYPE", "connection", "show"],
+      capture_output=True, text=True, check=True
+    )
     
-    # Check if the managed section exists
-    pattern = r'# BEGIN Go Note Go managed WiFi networks\n.*?# END Go Note Go managed WiFi networks'
-    if re.search(pattern, config, re.DOTALL):
-      # Replace the existing managed section
-      new_config = re.sub(pattern, managed_config, config, flags=re.DOTALL)
-    else:
-      # Append the managed section to the end of the file
-      new_config = config.rstrip() + "\n\n" + managed_config + "\n"
+    # Filter connections that are wifi and have names matching our managed networks
+    connections = []
+    networks = get_networks()
+    managed_ssids = [network['ssid'] for network in networks]
     
-    # Write the new config
-    with open('/tmp/wpa_supplicant.conf', 'w') as f:
-      f.write(new_config)
+    for line in result.stdout.splitlines():
+      parts = line.split(':', 1)
+      if len(parts) == 2:
+        name, conn_type = parts
+        if conn_type == 'wifi' and name in managed_ssids:
+          connections.append(name)
+    
+    return connections
+  except subprocess.CalledProcessError as e:
+    print(f"Error getting NetworkManager connections: {e}")
+    return []
+
+
+def add_wpa_connection(ssid, password):
+  """Add a new WPA secured WiFi connection."""
+  try:
+    conn_name = ssid
     
-    # Use sudo to copy the file to the correct location
-    shell('sudo cp /tmp/wpa_supplicant.conf /etc/wpa_supplicant/wpa_supplicant.conf')
-    shell('rm /tmp/wpa_supplicant.conf')
+    cmd = [
+      "sudo", "nmcli", "connection", "add",
+      "type", "wifi",
+      "con-name", conn_name,
+      "ifname", "wlan0",
+      "ssid", ssid,
+      "wifi-sec.key-mgmt", "wpa-psk",
+      "wifi-sec.psk", password
+    ]
     
+    subprocess.run(cmd, check=True, capture_output=True)
     return True
-  except Exception as e:
-    print(f"Error updating wpa_supplicant.conf: {e}")
+  except subprocess.CalledProcessError as e:
+    print(f"Error adding WPA connection for {ssid}: {e}")
+    print(f"Error output: {e.stderr}")
+    return False
+
+
+def add_open_connection(ssid):
```

Comment:
probably better to combine add_open_connection and add_wpa_connection since they're so similar